include/wifi/conn_mgr.h

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+#ifndef INCLUDE_WIFI_CONN_MGR_H_
+#define INCLUDE_WIFI_CONN_MGR_H_
+
+#include <zephyr/zephyr.h>
+#include <zephyr/net/wifi_mgmt.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @file conn_mgr.h
+ * 
+ * @defgroup Wi-Fi Connection Manager
+ * 
+ * @{
+ * 
+ * @brief Public APIs for the Wi-Fi Connection Manager
+ * 
+ */
+
+/**
+ * @brief Function for adding a credential to the list of networks
+ *      that the device will try to connect to. The maximum number of
+ *      credentials that can be programmed is configured by setting the
+ *      value of CONFIG_WIFI_CONN_MGR_SUPPORTED_CREDS in configuration.
+ * 
+ * @param new_param Credentials for a new network
+ * 
+ * @retval 0 if successful
+ * @retval -ENOMEM if all credentials have been configured
+ */
+int wifi_mgr_add_credential(struct wifi_connect_req_params *new_param);
+
+/**
+ * @brief Function to block a thread until the device has connected to
+ *      a Wi-Fi network. 
+ * 
+ *      NOTE: Currently only one thread can be blocked at a time with this function.
```

Comment:
Inspired by the link controller for the nRF91 series: Could it be an alternative to use an [iterable section](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/kernel/iterable_sections/index.html) to define a callback when a wifi connection is established? It would be something similar as `LTE_LC_ON_CFUN` (see https://github.com/nrfconnect/sdk-nrf/blob/main/include/modem/lte_lc.h#L840 and https://github.com/nrfconnect/sdk-nrf/blob/main/lib/lte_link_control/lte_lc.c#L1395).

The idea is that each library or thread that is waiting for a wifi connection/disconnection/... can use a macro, e.g. `WIFI_ON_CONN_EVT` to register a callback function that is called when the wifi-connection state is changed. This callback could e.g. give a semaphore to the waiting thread if the state is connected.